docs/docsite/rst/conf.py

```diff
@@ -228,7 +228,8 @@
 html_context = {
     'display_github': 'True',
     'show_sphinx': False,
-    'is_eol': False,
+    # Change First false to true for core EOL, 2nd false to true for package EOL
+     'is_eol': False if tags.has('core') else False,
```

Comment:
Why can't you just add an additional tag and set it on the calling side? This would give you more control/flexibility with less hardcoding.
```suggestion
     'is_eol': tags.has('is_eol'),
```

Reply:
Thanks @webknjaz ! I'm trying to understand how this would work. So once a repo goes EOL, it's always published as eol. So instead of having one place here in the conf.py to set the eol flag, I'd have to set it in multiple places around https://github.com/ansible/ansible-documentation/blob/devel/docs/docsite/Makefile#L104 (the calling side) plus the nox commands?

Reply:
If my understanding is correct, once the suggestion from @webknjaz is merged then you would need to update two places in the Makefile to EOL the package docs:

https://github.com/ansible/ansible-documentation/blob/3b43d6467dc97524a6e5184b069f4a590e9b4392/docs/docsite/Makefile#L104

and

https://github.com/ansible/ansible-documentation/blob/3b43d6467dc97524a6e5184b069f4a590e9b4392/docs/docsite/Makefile#L114

For example, currently that the `html` target is this:

```
CPUS=$(CPUS) $(MAKE) -f Makefile.sphinx 'DOCS_VARIANTS=-t ansible' html
```

To EOL the package docs, you would update that line to something like this:

```
CPUS=$(CPUS) $(MAKE) -f Makefile.sphinx 'DOCS_VARIANTS=-t ansible -t is_eol' html
```

I don't think we would need to touch the nox commands at this stage. They only call the Makefile.

Ofc you'd need to then make other updates to the Makefile separately to EOL the core docs when the time comes. So maybe it's a little error-prone to do that. I do like the idea of having a central config where you can switch eol on or off.

Maybe it would be possible to add some logic to the version helper somehow: https://github.com/ansible/ansible-documentation/blob/devel/docs/docsite/version_helper.py

Having a list of EOL versions somewhere could work. From the maintainer perspective, adding a version to an EOL list at the appropriate time seems like a natural, intuitive approach. Just a thought.

Reply:
@oraNod that config should probably live where make targets are being called, externally to the Makefile.

Reply:
@samccann in general, the idea is that this is not to be hardcoded in the docs generator but parametrized. And those params would govern how this flag is to be computed.